executor/src/constant_evaluator/mod.rs

```diff
@@ -134,16 +134,18 @@ impl<'a> Evaluator<'a> {
             BinaryOperator::Mul => left * right,
             BinaryOperator::Div => left.integer_div(right),
             BinaryOperator::Pow => left.pow(right.to_integer()),
-            BinaryOperator::Mod => (left.to_integer() % right.to_integer()).into(),
+            BinaryOperator::Mod => {
+                (left.to_arbitrary_integer() % right.to_arbitrary_integer()).into()
```

Comment:
Weren't these changes already present in the other PR?

Reply:
No, I realised in this one